src/backend/gl/src/device.rs

```diff
@@ -468,16 +469,74 @@ pub(crate) unsafe fn set_sampler_info<SetParamFloat, SetParamFloatVec, SetParamI
 impl d::Device<B> for Device {
     unsafe fn allocate_memory(
         &self,
-        _mem_type: c::MemoryTypeId,
+        mem_type: c::MemoryTypeId,
         size: u64,
     ) -> Result<n::Memory, d::AllocationError> {
-        // TODO
-        Ok(n::Memory {
-            properties: memory::Properties::CPU_VISIBLE | memory::Properties::CPU_CACHED,
-            first_bound_buffer: Cell::new(None),
-            size,
-            emulate_map_allocation: RefCell::new(std::ptr::null_mut()),
-        })
+        if (1 << mem_type.0) & IMAGE_MEM_TYPE_MASK != 0 {
```

Comment:
should this just be a comparison `1 << mem_type.0  == IMAGE_MEM_TYPE_MASK`?

Reply:
For this case, yes, but that assumes that there is only one type of image memory.  For the other classes there are multiple types of memory that could work, so the intention was to see whether the given type was in the bit mask at all.